Detectors/GlobalTrackingWorkflow/tpcinterpolationworkflow/src/TPCResidualWriterSpec.cxx

```diff
@@ -0,0 +1,81 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @file  TPCResidualWriterSpec.cxx
+
+#include <vector>
+
+#include "TTree.h"
+
+#include "Framework/ControlService.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "SpacePoints/TrackInterpolation.h"
+#include "TPCInterpolationWorkflow/TPCResidualWriterSpec.h"
+
+namespace o2
+{
+namespace tpc
+{
+
+void ResidualWriterTPC::init(InitContext& ic)
+{
+  mOutFileName = ic.options().get<std::string>("residuals-outfile");
+}
+
+void ResidualWriterTPC::run(ProcessingContext& pc)
+{
+  if (mFinished) {
+    return;
+  }
+
+  TFile outf(mOutFileName.c_str(), "recreate");
+  if (outf.IsZombie()) {
+    LOG(FATAL) << "Failed to open output file " << mOutFileName;
+  }
+
+  TTree tree(mTreeName.c_str(), "Tree of TPC residuals and reference tracks");
+  auto tracks = std::move(pc.inputs().get<const std::vector<TrackData>>("tracks"));
+  auto tracksPtr = &tracks;
+  auto residuals = std::move(pc.inputs().get<const std::vector<TPCClusterResiduals>>("residuals"));
+  auto residualsPtr = &residuals;
+  tree.Branch(mOutTracksBranchName.c_str(), &tracksPtr);
```

Comment:
in view of the previous comment, it is better to avoid ``branch->Fill()`` calls and work on the tree level.